src/server/project.ts

```diff
@@ -1322,6 +1407,70 @@ namespace ts.server {
         refreshDiagnostics() {
             this.projectService.sendProjectsUpdatedInBackgroundEvent();
         }
+
+        /*@internal*/
+        getPackageJsonsVisibleToFile(fileName: string, rootDir?: string): readonly PackageJsonInfo[] {
+            const packageJsonCache = this.packageJsonCache;
+            const watchPackageJsonFile = this.watchPackageJsonFile.bind(this);
+            const currentDirectory = this.currentDirectory;
+            const getCanonicalFileName = this.getCanonicalFileName;
+            const result: PackageJsonInfo[] = [];
+            const rootPath = rootDir && toPath(rootDir, currentDirectory, this.getCanonicalFileName);
+            forEachAncestorDirectory(getDirectoryPath(fileName), function processDirectory(directory): boolean | undefined {

```

Comment:
Is there a reason for using functionExpression instead of arrow here?

Reply:
Yes, it recurses  a few lines below.